app/backend/src/couchers/servicers/account.py

```diff
@@ -118,11 +120,16 @@ def ChangeEmail(self, request, context):
         with session_scope() as session:
             user = session.query(User).filter(User.id == context.user_id).one()
 
-            # otherwise we're good
             user.new_email = request.new_email
-            token, expiry_text = set_email_change_token(session, user)
+            if user.hashed_password is None:
```

Comment:
We should probably write a `@property` helper for `user.has_password`, it's a bit ugly to use this everywhere

Reply:
I can add that